server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2086,6 +2089,18 @@ public boolean expunge(UserVmVO vm, long callerUserId, Account caller) {
         }
     }
 
+    /**
+     * Release network resources, it was done on vm stop previously.
+     * @param id vm id
+     * @throws ConcurrentOperationException
+     * @throws ResourceUnavailableException
+     */
+    private void releaseNetworkResourcesOnExpunge(long id) throws ConcurrentOperationException, ResourceUnavailableException {
+        final VMInstanceVO vmInstance = _vmDao.findById(id);
```

Comment:
It is possible for ``_vmDao.findById`` to return ``null``.  Please handling for that case.